File: src/test/java/fr/deroffal/aoc/Day01Test.java
package fr.deroffal.aoc;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class Day01Test {

	@Test
	void input1122_offset1_retourne_3() {
		assertEquals(3, Day01.calculerScore("1122", 1));
	}

	@Test
	void input1111_offset1_retourne_4() {
		assertEquals(4, Day01.calculerScore("1111", 1));
	}

	@Test
	void input1234_offset1_retourne_0() {
		assertEquals(0, Day01.calculerScore("1234", 1));
	}

	@Test
	void input91212129_offset1_retourne_9() {
		assertEquals(9, Day01.calculerScore("91212129", 1));
	}

	@Test
	void input1212_offset2_retourne_6() {
		assertEquals(6, Day01.calculerScore("1212", 2));
	}

	@Test
	void input1221_offset2_retourne_0() {
		assertEquals(0, Day01.calculerScore("1221", 2));
	}

	@Test
	void input123425_offset3_retourne_4() {
		assertEquals(4, Day01.calculerScore("123425", 3));
	}

	@Test
	void input123123_offset3_retourne_12() {
		assertEquals(12, Day01.calculerScore("123123", 3));
	}

	@Test
	void input12131415_offset4_retourne_4() {
		assertEquals(4, Day01.calculerScore("12131415", 4));
	}

}

File: kotlin/src/test/kotlin/fr/deroffal/aoc/kotlin/Day01Test.kt
package fr.deroffal.aoc.kotlin

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.aggregator.ArgumentsAccessor
import org.junit.jupiter.params.provider.CsvSource

internal class Day01Test {

    private val day01 = Day01()

    @CsvSource(
            "3,1122",
            "4,1111",
            "0,1234",
            "9,91212129"
    )
    @ParameterizedTest(name = "Input ''{1}'' gives captcha {0}")
    fun `part1 matches examples`(arguments: ArgumentsAccessor) {
        val expected = arguments.getInteger(0)
        val input = arguments.getString(1)
        assertEquals(expected, day01.solvePart1(input))
    }

    @CsvSource(
            "6,1212",
            "0,1221",
            "4,123425",
            "12,123123",
            "4,12131415"
    )
    @ParameterizedTest(name = "Input ''{1}'' gives captcha {0}")
    fun `part2 matches examples`(expected: Int, input: String) {
        assertEquals(expected, day01.solvePart2(input))
    }
}